swagger_parser/lib/src/parser/model/universal_component_class.dart

```diff
@@ -35,6 +38,13 @@ final class UniversalComponentClass extends UniversalDataClass {
     Map<String, List<UniversalType>> refProperties,
   })? discriminator;
 
+  /// When using a discriminated oneOf, where this class is one of the discriminated values, this field contains the information about the parent
+  ({
+    // The name of the property that is used to discriminate the oneOf variants
+    String propertyValue,
+    String parentClass,
+  })? discriminatorValue;
```

Comment:
This `discriminatorValue` field could potentially be combined with `discriminator` above, perhaps adding an extra field to denote whether this class is a discrinator parent or discriminated value class.